src/main/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryListFilterChainBuilder.java

```diff
@@ -258,12 +260,31 @@ public ItineraryListFilterChainBuilder withFares(FareService fareService) {
     return this;
   }
 
+  public ItineraryListFilterChainBuilder withRemoveTimeshiftedItinerariesWithSameRoutesAndStops(
+    boolean remove
+  ) {
+    this.removeTimeshiftedItinerariesWithSameRoutesAndStops = remove;
+    return this;
+  }
+
   @SuppressWarnings("CollectionAddAllCanBeReplacedWithConstructor")
   public ItineraryListFilterChain build() {
     List<ItineraryListFilter> filters = new ArrayList<>();
 
     filters.addAll(buildGroupByTripIdAndDistanceFilters());
 
+    if (removeTimeshiftedItinerariesWithSameRoutesAndStops) {
+      filters.add(
+        new GroupByFilter<>(
+          GroupBySameRoutesAndStops::new,
+          List.of(
+            new SortingFilter(generalizedCostComparator()),
```

Comment:
Do you want to keep the lowest cost or the earliest departure?

Reply:
I want to keep the earliest departure.

Reply:
I think this should be the earliest itinerary for a depart-after search, and the latest for a arrive-by search?

Reply:
Could you give me a hint how to achieve that?

Reply:
Should I be passing the arriveBy flag through to the chain and then the the ListSection to head or tail respectively?

Reply:
I this this would work: `new SortingFilter(SortOrderComparator.comparator(sortOrder))`. Strictly, this will sort street itineraries by them selves - but I guess we do not care in this case, so it is ok.


Can you move the `new GroupByFilter<>(...)` into a builder method as well, like the `buildGroupByTripIdAndDistanceFilters()`?

Reply:
Done for both.

Reply:
Sorry, it's really fixed now.

Reply:
Sorry, now it's really fixed.